From 68f4f9045feddd8b8eac81c01ddbcfac4017a297 Mon Sep 17 00:00:00 2001
From: Jakub Nowacki <jakubx.nowacki@intel.com>
Date: Tue, 4 Jul 2023 15:54:25 +0200
Subject: [PATCH] Add telemetry triggers to the message registry

This commit adds Redfish event log message entries for telemetry service
triggers. It adds 5 message entries:
1. TriggerNumericWarning
2. TriggerNumericCritical
3. TriggerDiscreteOK
4. TriggerDiscreteWarning
5. TriggerDiscreteCritical

Tested:
On platform - messages appear as expected on Redfish log for corresponding actions.

Signed-off-by: Jakub Nowacki <jakubx.nowacki@intel.com>
---
 redfish-core/include/registries.hpp           |  2 +-
 .../registries/openbmc_message_registry.hpp   | 60 ++++++++++++++++++-
 2 files changed, 60 insertions(+), 2 deletions(-)

diff --git a/redfish-core/include/registries.hpp b/redfish-core/include/registries.hpp
index bad31c43..d8667e3a 100644
--- a/redfish-core/include/registries.hpp
+++ b/redfish-core/include/registries.hpp
@@ -50,7 +50,7 @@ struct Message
     const char* message;
     const char* messageSeverity;
     const size_t numberOfArgs;
-    std::array<const char*, 5> paramTypes;
+    std::array<const char*, 6> paramTypes;
     const char* resolution;
 };
 using MessageEntry = std::pair<const char*, const Message>;
diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index 5fae6619..64533135 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -63,7 +63,7 @@
 namespace redfish::registries::openbmc
 {
 const Header header = {
-    "Copyright 2022 OpenBMC. All rights reserved.",
+    "Copyright 2023 OpenBMC. All rights reserved.",
     "#MessageRegistry.v1_4_0.MessageRegistry",
     "OpenBMC.0.4.0",
     "OpenBMC Message Registry",
@@ -2387,6 +2387,64 @@ constexpr std::array registry = {
             {},
             "None.",
         }},
+    MessageEntry{
+        "TriggerNumericWarning",
+        {
+            "Indicates that numeric trigger threshold is crossed.",
+            "Numeric threshold '%1' of trigger '%2' is crossed on sensor %3, "
+            "recorded value: %4, crossing direction: %5, timestamp: %6",
+            "Warning",
+            6,
+            {"string", "string", "string", "number", "string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "TriggerNumericCritical",
+        {
+            "Indicates that numeric trigger threshold is crossed and may "
+            "require attention.",
+            "Numeric threshold '%1' of trigger '%2' is crossed on sensor %3, "
+            "recorded value: %4, crossing direction: %5, timestamp: %6",
+            "Critical",
+            6,
+            {"string", "string", "string", "number", "string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "TriggerDiscreteOK",
+        {
+            "Indicates that discrete trigger condition is met.",
+            "Discrete condition '%1' of trigger '%2' is met on sensor %3, "
+            "recorded value: %4, timestamp: %5",
+            "OK",
+            5,
+            {"string", "string", "string", "string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "TriggerDiscreteWarning",
+        {
+            "Indicates that discrete trigger condition is met and may require "
+            "attention.",
+            "Discrete condition '%1' of trigger '%2' is met on sensor %3, "
+            "recorded value: %4, timestamp: %5",
+            "Warning",
+            5,
+            {"string", "string", "string", "string", "string"},
+            "None.",
+        }},
+    MessageEntry{
+        "TriggerDiscreteCritical",
+        {
+            "Indicates that discrete trigger condition is met and requires "
+            "immediate attention.",
+            "Discrete condition '%1' of trigger '%2' is met on sensor %3, "
+            "recorded value: %4, timestamp: %5",
+            "Critical",
+            5,
+            {"string", "string", "string", "string", "string"},
+            "None.",
+        }},
     MessageEntry{
         "VoltageRegulatorOverheated",
         {
-- 
2.25.1

